patroni/utils.py

```diff
@@ -342,3 +343,17 @@ def split_host_port(value, default_port):
     t = value.rsplit(':', 1)
     t.append(default_port)
     return t[0], int(t[1])
+
+
+def uri(proto, netloc, path='', user=None):
+    host, port = netloc if isinstance(netloc, (list, tuple)) else split_host_port(netloc, 0)
+    if host and host[0] != '[' and host[-1] != ']':
+        try:
+            socket.inet_pton(socket.AF_INET6, host)
```

Comment:
Looks like Windows support for `inet_pton` was added only in 3.4, which makes supported versions a bit different.